docs/develop/create-fexpr.rst

```diff
@@ -0,0 +1,391 @@
+
+.. code-block:: python
+
+    import pytest
+    import random
+    from datatable import dt, f, gcd
+    from tests import assert_equals  # checks equality of Frames
+    from math import gcd as math_gcd
+
+    def test_equal_columns():
+        DT = dt.Frame(A=[1, 2, 3, 4, 5])
+        RES = DT[:, gcd(f.A, f.A)]
+        assert_equals(RES, dt.Frame([1, 1, 1, 1, 1]/dt.int32))
+
+    @pytest.mark.parametrize("seed", [random.getrandbits(63)])
+    def test_random(seed):
+        random.seed(seed)
+        n = 100
+        src1 = [random.randint(1, 1000) for i in range(n)]
+        src2 = [random.randint(1, 100) for i in range(n)]
+        DT = dt.Frame(A=src1, B=src2)
+        RES = DT[:, gcd(f.A, f.B)]
+        assert_equals(RES, dt.Frame([math_gcd(src1[i], src2[i])
+                                     for i in range(n)]))
+
+When writing tests try to test any corner cases that you can think of. For
+example, what if one of the numbers is 0? Negative? Add tests for various
+column types, including invalid ones.
+
+
+Documentation
+-------------
+
+The final piece of the puzzle is the documentation. We've already written the
+documentation for our function: the ``doc_gcd`` variable declared earlier.
+However, for now this is only visible from python when you run ``help(gcd)``.
+We also want the documentation to be visible in our official readthedocs
+website, which requires a few more steps. So:
+
+First, create file ``docs/api/dt/gcd.rst``. The content of the file should
+contain just few lines:
+
+.. code-block:: rst
+
+    .. py:currentmodule:: datatable
+
+    .. xfunction:: datatable.gcd
+        :doc: src/core/fexpr/fexpr_gcd.cc doc_gcd
+        :src: src/core/fexpr/fexpr_gcd.cc py_gcd
+        :tests: tests/expr/test-gcd.py
+
+In these lines we declare: in which source file the docstring can be found,
+and what is the name of its variable. The documentation generator will be
+looking for a ``static const char* doc_gcd`` variable in the source. Then
+we also declare the name of the function which provides the gcd functionality.
+The generator will look for a function with that name in the specified source
+file and create a link to that source in the output doc file. Lastly, the
+``:tests:`` parameter says which file contains tests dedicated to this
+function, this will also become a link in the generated documentation.
+
+This RST file now needs to be added to the toctree: open the file
+``docs/api/index-api.rst`` and add it into the ``.. toctree::`` list at the
+bottom, and also add it to the table of all functions.
+
+Lastly, open ``docs/releases/v{LATEST}.rst`` (this is our changelog) and write
+a brief paragraph about the new function:
+
+.. code-block:: rst
+
+    Frame
+    -----
+    ...
+
+    -[new] Added new function :func:`gcd()` to compute the greatest common
+      divisor of two columns. [#1234]
+
+The ``[#1234]`` is a link to the GitHub issue where the ``gcd()`` function
```

Comment:
I would spell it as `[#NNNN]` to avoid confusion with the existing `1234` issue/PR.

Reply:
fixed